src/NuGet.Protocol.Core.v3/HttpHandlerResourceV3Provider.cs

```diff
@@ -66,12 +64,7 @@ private HttpClientHandler TryGetCredentialAndProxy(PackageSource packageSource)
                 && !String.IsNullOrEmpty(packageSource.UserName)
                 && !String.IsNullOrEmpty(packageSource.Password))
             {
-                var cache = new CredentialCache();
```

Comment:
Talking to the VSO folks, this will break some of their scenarios where the hosts are different between where index.json is hosted and where the registration blobs are at.

The suggestion is to make the code flow the credentials around. @zarenner is going to propose changes, and I'm going to ask @blowdart to take a look to see if we are worried about credential leak or not

@daazose